packages/squiggle-lang/src/rescript/ReducerProject/ReducerProject.res

```diff
@@ -0,0 +1,397 @@
+    environment: getEnvironment(this),
+  }
+
+  let doRunWithContinuation = (
+    this: t,
+    sourceId: string,
+    continuation: ProjectItem.T.continuation,
+  ): unit => {
+    let accessors = buildProjectAccessors(this)
+    let newItem = this->getItem(sourceId)->ProjectItem.run(continuation, accessors)
+    Belt.Map.String.set(this["items"], sourceId, newItem)->T.Private.setFieldItems(this, _)
+    setContinuation(this, sourceId, accessors.continuation)
+  }
+
+  type runState = (ProjectItem.T.resultArgumentType, ProjectItem.T.continuation)
+
+  let tryRunWithContinuation = (
+    this: t,
+    sourceId: string,
+    (rPrevResult: ProjectItem.T.resultArgumentType, continuation: ProjectItem.T.continuation),
+  ): (ProjectItem.T.resultArgumentType, ProjectItem.T.continuation) => {
+    switch getResult(this, sourceId) {
+    | Some(result) => (result, getContinuation(this, sourceId)) // already ran
+    | None =>
+      switch rPrevResult {
+      | Error(error) => {
+          setResult(this, sourceId, Error(error))
+          (Error(error), continuation)
+        }
+      | Ok(_prevResult) => {
+          doRunWithContinuation(this, sourceId, continuation)
+          (
+            getResult(this, sourceId)->Belt.Option.getWithDefault(rPrevResult),
+            getContinuation(this, sourceId),
+          )
+        }
+      }
+    }
+  }
+
+  let runAll = (this: t): unit => {
+    let runOrder = Topology.getRunOrder(this)
+    let initialState = (Ok(InternalExpressionValue.IEvVoid), getStdLib(this))
+    let _finalState = Belt.Array.reduce(runOrder, initialState, (currState, currId) =>
+      tryRunWithContinuation(this, currId, currState)
+    )
+  }
+
+  let run = (this: t, sourceId: string): unit => {
+    let runOrder = Topology.getRunOrderFor(this, sourceId)
+    let initialState = (Ok(InternalExpressionValue.IEvVoid), getStdLib(this))
+    let _finalState = Belt.Array.reduce(runOrder, initialState, (currState, currId) =>
+      tryRunWithContinuation(this, currId, currState)
+    )
+  }
+
+  let evaluate = (sourceCode: string) => {
+    let project = createProject()
+    setSource(project, "main", sourceCode)
+    runAll(project)
+    (
+      getResult(project, "main")->Belt.Option.getWithDefault(IEvVoid->Ok),
+      getContinuation(project, "main"),
+    )
+  }
+}
+
+/*
+  PUBLIC FUNCTIONS
+*/
+
+/*
+Creates a new project to hold the sources, executables, bindings, and other data. 
+The new project runs the sources according to their topological sorting because of the includes and continues.
+
+Any source can include or continue other sources. "Therefore, the project is a graph data structure."
+The difference between including and continuing is that includes are stated inside the source code while continues are stated in the project.
+
+To run a group of source codes and get results/bindings, the necessary methods are
+- setSource
+- setContinues
+- parseIncludes
+- run or runAll
+- getExternalBindings
+- getExternalResult
+*/
```

Comment:
Could you add a test or something that shows this full pipeline? Like, it has 3-4 separate Squiggle files with imports, loads them all, then runs calls on them, with the expected results? 

It looks like the parser also changed, but it's difficult to understand just how or what it means.